google/cloud/spanner/value.cc

```diff
@@ -535,6 +560,22 @@ StatusOr<std::int64_t> Value::GetValue(std::int64_t,
   return x;
 }
 
+StatusOr<float> Value::GetValue(float, google::protobuf::Value const& pv,
+                                google::spanner::v1::Type const&) {
+  if (pv.kind_case() == google::protobuf::Value::kNumberValue) {
+    return static_cast<float>(pv.number_value());  // narrowing
+  }
+  if (pv.kind_case() != google::protobuf::Value::kStringValue) {
+    return Status(StatusCode::kUnknown, "missing FLOAT32");
+  }
+  std::string const& s = pv.string_value();
+  auto const inf = std::numeric_limits<float>::infinity();
+  if (s == "-Infinity") return -inf;
```

Comment:
Just to show that I fear floating point, but I don't understand it: I think C++ does not  guarantee that `- std::numeric_limits<float>::infinity()` yields a valid value, or that `::infinity()` even exists. OTOH, on the platforms we could possibly care about, those things don't matter.

Reply:
We could try to (statically?) assert `std::numeric_limits<float>::has_infinity()` somewhere, or otherwise require IEEE 754 support.  Otherwise, I believe we'll at least get `HUGE_VALF` from `infinity()`, and so use it for "Infinity" on such a platform.

Reply:
Optional: add the  static assert.  But it could get repetitive if we do it in each place we use that assumption, your call.

Reply:
Done (in a single place).  Also added one for `double`.